flink/v1.19/flink/src/main/java/org/apache/iceberg/flink/sink/shuffle/AggregatedStatistics.java

```diff
@@ -19,53 +19,87 @@
 package org.apache.iceberg.flink.sink.shuffle;
 
 import java.io.Serializable;
-import org.apache.flink.api.common.typeutils.TypeSerializer;
+import java.util.Arrays;
+import java.util.Map;
+import org.apache.iceberg.SortKey;
 import org.apache.iceberg.relocated.com.google.common.base.MoreObjects;
-import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.relocated.com.google.common.base.Objects;
 
 /**
  * AggregatedStatistics is used by {@link DataStatisticsCoordinator} to collect {@link
  * DataStatistics} from {@link DataStatisticsOperator} subtasks for specific checkpoint. It stores
  * the merged {@link DataStatistics} result from all reported subtasks.
  */
-class AggregatedStatistics<D extends DataStatistics<D, S>, S> implements Serializable {
-
+class AggregatedStatistics implements Serializable {
   private final long checkpointId;
-  private final DataStatistics<D, S> dataStatistics;
-
-  AggregatedStatistics(long checkpoint, TypeSerializer<DataStatistics<D, S>> statisticsSerializer) {
-    this.checkpointId = checkpoint;
-    this.dataStatistics = statisticsSerializer.createInstance();
-  }
+  private final StatisticsType type;
+  private final Map<SortKey, Long> keyFrequency;
```

Comment:
combine both Map and Sketch stats in the same aggregated statistics object would allow run-time switch from `Map` stats to `Sketch`.